backend/app/commands/DeleteResource.scala

```diff
@@ -16,16 +16,18 @@ import scala.concurrent.ExecutionContext
 class DeleteResource( manifest: Manifest, index: Index, previewStorage: ObjectStorage, objectStorage: ObjectStorage)  (implicit ec: ExecutionContext)
    extends Timing {
 
-     private def deletePagePreviewObjects(s3Objects: List[String]): Attempt[List[Unit]] = {
-        logger.info(s"Deleting objects: ${s3Objects.mkString(", ")}")
-       val deletionAttempts = s3Objects
-         .map(previewStorage.delete)
-         .map(Attempt.fromEither)
+     private def deleteFromS3Preview(blobUri: Uri, pagePreviewKeys: Set[String]): Attempt[Iterator[Unit]] = {
+       // The full-document preview, as well as all the previews of individual pages
+       val keys = pagePreviewKeys + blobUri.toStoragePath
+       logger.info(s"Deleting ${keys.size} objects from preview storage")
 
-       Attempt.sequence(deletionAttempts)
+       // Group, just in case we have thousands of pages
+       Attempt.traverse(keys.grouped(500)) { batchOfS3Keys =>
+         Attempt.fromEither(previewStorage.deleteMultiple(batchOfS3Keys))
```

Comment:
Lovely batching! The [AWS API docs](https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObjects.html) say that 1000 object ids is the limit for a batch, I think we get a lot of benefit with a chunk size of 500.